lib/active_model/serializer/attribute.rb

```diff
@@ -0,0 +1,13 @@
+module ActiveModel
+  class Serializer
+    Attribute = Struct.new(:name, :key, :block) do
+      def value(serializer)
+        if block
+          serializer.instance_eval(&block)
+        else
+          serializer.read_attribute_for_serialization(name)
+        end
+      end
```

Comment:
Agreed. Every reference to an `Attribute` object is currently marked as `@api private`.